pkg/maps/ctmap/ctmap.go

```diff
@@ -120,31 +124,37 @@ func InitMapInfo(tcpMaxEntries, anyMaxEntries int) {
 	natV4 := natMaps[0]
 	natV6 := natMaps[1]
 
-	mapType := MapTypeIPv4TCPLocal
-	for _, maxEntries := range []int{MapNumEntriesLocal, tcpMaxEntries} {
-		setupMapInfo(MapType(mapType), "CT_MAP_TCP4",
-			&CtKey4{},
-			int(unsafe.Sizeof(CtKey4{})), maxEntries,
-			bpf.ConvertKeyValue, natV4)
-		mapType++
-		setupMapInfo(MapType(mapType), "CT_MAP_TCP6",
-			&CtKey6{},
-			int(unsafe.Sizeof(CtKey6{})), maxEntries,
-			bpf.ConvertKeyValue, natV6)
-		mapType++
-	}
-	for _, maxEntries := range []int{MapNumEntriesLocal, anyMaxEntries} {
-		setupMapInfo(MapType(mapType), "CT_MAP_ANY4",
-			&CtKey4{},
-			int(unsafe.Sizeof(tuple.TupleKey4{})), maxEntries,
-			bpf.ConvertKeyValue, natV4)
-		mapType++
-		setupMapInfo(MapType(mapType), "CT_MAP_ANY6",
-			&CtKey6{},
-			int(unsafe.Sizeof(CtKey6{})), maxEntries,
-			bpf.ConvertKeyValue, natV6)
-		mapType++
-	}
+	setupMapInfo(MapType(MapTypeIPv4TCPLocal), "CT_MAP_TCP4",
+		&CtKey4{}, int(unsafe.Sizeof(CtKey4{})), &CtEntry{}, int(unsafe.Sizeof(CtEntry{})),
+		MapNumEntriesLocal, bpf.ConvertKeyValue, natV4)
```

Comment:
Small nit: could we simplify the setupMapInfo() as well a bit e.g. we pass in `bpf.ConvertKeyValue` everywhere. Couldn't we also derive the key and value size based on the concrete types we pass in there?

Reply:
> could we simplify the setupMapInfo() as well a bit e.g. we pass in bpf.ConvertKeyValue everywhere.

yes

>Couldn't we also derive the key and value size based on the concrete types we pass in there?

no, if the function arguments would have `func f(foo interface{})` and if we pass `f(CtEntry{}` the sizeof an interface in golang.